### PR TITLE
[INFRA] Remove incubator/incubating for graduation

### LINKED ISSUE #2415: [Umbrella] Apache Celeborn Graduation Tasks
### Describe the proposal
The ASF board has approved a resolution to graduate Celeborn into a full Top Level Project. Thanks to everyone for your help to get to this point.

To transition from the Apache Incubator to a new TLP, there's a few action items we need to do to complete the transition. These are identified by https://incubator.apache.org/guides/transferring.html#life_after_graduation.

### Task list

- Update Incubator status records
  - [x] Update the incubator status file (in svn)
    https://svn.apache.org/repos/asf/incubator/public/trunk/content/projects/celeborn.xml
  - [x] Update podling status page
    https://incubator.apache.org/projects/
    https://incubator.apache.org/projects/celeborn.html
    https://svn.apache.org/repos/asf/incubator/public/trunk/content/podlings.xml
- Source repo changes
  - [x] Create JIRA ticket INFRA-25635 to request ASF Infra remove "incubator-" prefix of git repos
    https://github.com/apache/incubator-celeborn
    https://github.com/apache/incubator-celeborn-website
  - [x] Create JIRA ticket INFRA-25636 to request ASF Infra move dist folder in SVN repositories from the incubator to other locations
    old: https://dist.apache.org/repos/dist/release/incubator/celeborn
    new: https://dist.apache.org/repos/dist/release/celeborn
  - [x] Post announcement to dev list telling everyone the repo is about repos renaming
  - [x] Post an announcement containing instructions for developers describing how to change their git remotes
    https://lists.apache.org/thread/5yhb6z8g1bvy5kptkf7b8loy28my8hnc
  - [x] Update website, pom.xml, and other resources to point to the new repository location
 Post an announcement containing instructions for developers describing how to change their git remotes.

 Update website, pom.xml, and other resources to point to the new repository location
- Source code
  - [x] Remove "Incubating" from source code
  - [x] Remove DISCLAIMER from git repo
  - [x] Update Release scripts and template

- Websites
  - [x] Remove incubating/podling related resources from the website
  - [x] Update the process for becoming a committer or joining PMC
 
- Ecosystem
  - [x] Remove Incubating from third party projects including Gluten.
     - [x] Gluten: https://github.com/apache/incubator-gluten/issues/5142.

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Remove incubator/incubating for graduation including:

- Remove `incubator`/`Incubating`.
- Remove `Podling`.
- Update `PPMC` to `PMC`.

Fix https://github.com/apache/celeborn/issues/2415.

### CODE REVIEW

File: docs/community/release_guide.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Celeborn Release Guide
 
 ## Introduction
 
-The Apache Celeborn (Incubating) project periodically declares and publishes releases. A release is one or more packages
+The Apache Celeborn project periodically declares and publishes releases. A release is one or more packages
```
Comment (FMX) [inline docs/community/release_guide.md:24]: This release guide can be changed into a TLP version. You can refer to [Kybbui's release guide](https://github.com/apache/kyuubi/blob/master/docs/contributing/code/release.md).

File: mkdocs.yml [hunk 2]
```diff
@@ -56,21 +56,9 @@ extra:
...
 
 copyright: >
-  <img src="https://www.apachecon.com/event-images/acna2022-wide-dark.png" style="height: 100px;" alt="ApacheCon North America">
-  <img src="https://incubator.apache.org/images/incubator_feather_egg_logo_bw_crop.png" style="height: 87px; margin: 0 0 10px 0;">
   <br>
-  Copyright © 2022 The Apache Software Foundation
+  Copyright © 2024 The Apache Software Foundation, Licensed under the Apache License, Version 2.0.
   <a href="https://privacy.apache.org/policies/privacy-policy-public.html">Privacy Policy<a/><br>
-  <br>
-  Apache Celeborn™, Apache Incubator, Apache, the Apache feather logo, and the Apache Incubator project logo are
```
Comment (pan3793) [inline mkdocs.yml:65]: should we retain this statement by just remove "Apache Incubator"'s part

File: .github/bin/build_docs.sh [hunk 3]
```diff
@@ -36,12 +36,12 @@ TAR_NAME=${REF_VERSION_NAME##*/}
 
 if [[ "$REF_VERSION_NAME" == *"-tm"* ]] ;then
     DIR_VERSION_NAME=$(echo "$TAR_NAME" | sed -r "s/v*(.*)\.tar\.gz/\1/g")
-    TAR_DIR_NAME=incubator-celeborn-$DOC_LINK_PATH
-    wget "https://github.com/apache/incubator-celeborn/releases/download/${REF_VERSION_NAME}"
+    TAR_DIR_NAME=celeborn-$DOC_LINK_PATH
+    wget "https://github.com/apache/celeborn/releases/download/${REF_VERSION_NAME}"
```
Comment (pan3793) [inline .github/bin/build_docs.sh:40]: there is hacky for `-tm` artifacts, we should either keep it, or update the GitHub Release artifacts in the main repo. (I prefer the latter)

Comment (SteNicholas) [reply .github/bin/build_docs.sh:40]: This could fix in #43.

File: mkdocs.yml [hunk 4]
```diff
@@ -56,20 +56,11 @@ extra:
...
   <br>
-  Apache Celeborn™, Apache Incubator, Apache, the Apache feather logo, and the Apache Incubator project logo are
-  trademarks or registered trademarks of The Apache Software Foundation.<br>
-  <br>
-  Apache Celeborn™ is an effort undergoing incubation at The Apache Software Foundation (ASF), sponsored by the
-  Apache Incubator. Incubation is required of all newly accepted projects until a further review indicates that
-  the infrastructure, communications, and decision making process have stabilized in a manner consistent with
-  other successful ASF projects. While incubation status is not necessarily a reflection of the completeness or
-  stability of the code, it does indicate that the project has yet to be fully endorsed by the ASF.<br>
+  Apache Celeborn™, Apache, the Apache feather logo are trademarks or registered trademarks of The Apache Software Foundation.<br>
```
Comment (pan3793) [inline mkdocs.yml]:
```suggestion
  Apache Celeborn™, Apache, and the Apache feather logo are trademarks or registered trademarks of The Apache Software Foundation.<br>
```

**Review by pan3793 — APPROVED** [review]

**Review by FMX — APPROVED** [review]
LGTM. Thanks. Merged.

**Pull request merged**